src/fiatExchanges/SelectProvider.tsx

```diff
@@ -0,0 +1,516 @@
+        <View style={styles.noQuotesContainer}>
+          <Text style={styles.noQuotesText}>
+            {t('selectProviderScreen.somePaymentsUnavailable')}{' '}
+            <Text onPress={openDialog} style={styles.underline}>
+              {t('selectProviderScreen.learnMore')}
+            </Text>
+          </Text>
+        </View>
+      )}
+      <Dialog
+        title={t('selectProviderScreen.whyMissingPayments')}
+        isVisible={isDialogVisible}
+        actionText={t('selectProviderScreen.dismiss')}
+        actionPress={dismissDialog}
+        isActionHighlighted={false}
+        onBackgroundPress={dismissDialog}
+      >
+        <Text style={styles.dialog}>{t('selectProviderScreen.missingPaymentsExplained')}</Text>
+      </Dialog>
+    </>
+  )
+}
+
+function ExchangesSection({
+  flow,
+  selectedCurrency,
+}: {
+  flow: CICOFlow
+  selectedCurrency: Currency
+}) {
+  const { t } = useTranslation()
+  const goToExchangesScreen = () => {
+    navigate(Screens.ExternalExchanges, {
+      currency: selectedCurrency,
+      isCashIn: flow === CICOFlow.CashIn,
+    })
+  }
+  return (
+    <View style={styles.container}>
+      <Touchable onPress={goToExchangesScreen}>
+        <View style={{ ...styles.expandableContainer, paddingVertical: 27 }}>
+          <View style={styles.left}>
+            <Text style={styles.category}>{t('selectProviderScreen.cryptoExchange')}</Text>
+
+            <Text style={styles.fee}>{t('selectProviderScreen.feesVary')}</Text>
+          </View>
+
+          <View style={styles.right}>
+            <Text style={styles.linkToOtherScreen}>{t('selectProviderScreen.viewExchanges')}</Text>
+          </View>
+        </View>
+      </Touchable>
+    </View>
+  )
+}
+
+function LocalProviderMobileMoneySection({
+  localProvider,
+  digitalAsset,
+}: {
+  localProvider: LocalCicoProvider | undefined
+  digitalAsset: CiCoCurrency
+}) {
+  const { t } = useTranslation()
+  if (!localProvider) {
+    return null
+  }
+  const goToLocalProvider = () => {
+    navigateToURI(localProvider[digitalAsset === CiCoCurrency.CUSD ? 'cusd' : 'celo'].url)
+  }
+  return (
+    <View style={styles.container}>
+      <Touchable onPress={goToLocalProvider}>
+        <View style={{ ...styles.expandableContainer, paddingVertical: 27 }}>
+          <View style={styles.left}>
+            <Text style={styles.category}>{t('selectProviderScreen.mobileMoney')}</Text>
+            <Text style={styles.fee}>{t('selectProviderScreen.feesVary')}</Text>
+          </View>
+
+          <View style={styles.right}>
+            <Text style={styles.linkToOtherScreen}>{localProvider.name}</Text>
+          </View>
+        </View>
+      </Touchable>
+    </View>
+  )
+}
+
+interface PaymentMethodSectionProps {
+  paymentMethod: PaymentMethod
+  cicoQuotes: CicoQuote[]
+  setNoPaymentMethods: React.Dispatch<React.SetStateAction<boolean>>
+  quoteOnPress: (cicoQuote: CicoQuote) => void
+}
+
+export function PaymentMethodSection({
```

Comment:
i find this file gigantic, i wonder if we can  make `PaymentMethodSection` its own component